### PR TITLE
add rails 4 compatibility

### PR DESCRIPTION
**Pull request by jebentier**
### Added
- Backwards compatibility with Rails 4

### CODE REVIEW

File: lib/active_record/mysql/enum/mysql_adapter.rb [hunk 1]
```diff
@@ -25,19 +27,35 @@ def native_database_types #:nodoc
         # be done on a per adapter basis, but is generalized here.
         #
         # will generate enum('a', 'b', 'c') for :limit => [:a, :b, :c]
-        def type_to_sql(type, limit: nil, precision: nil, scale: nil, unsigned: nil, **) # :nodoc:
-          if type.to_s == 'enum'
-            native = native_database_types[type]
-            column_type_sql = (native || {})[:name] || 'enum'
+        if Rails::VERSION::MAJOR < 5
+          def type_to_sql(type, limit = nil, precision = nil, scale = nil, unsigned = nil, **) # :nodoc:
```
Comment (ColinDKelley) [inline lib/active_record/mysql/enum/mysql_adapter.rb]: Maybe `**_options` to be clear that those options are being ignored?

File: lib/active_record/mysql/enum/mysql_adapter.rb [hunk 2]
```diff
@@ -25,19 +27,35 @@ def native_database_types #:nodoc
...
-            column_type_sql
-          else
-            super(type, limit: limit, precision: precision, scale: scale, unsigned: unsigned)
+              column_type_sql
+            else
+              super(type, limit, precision, scale, unsigned)
+            end
+          end
+        else
+          def type_to_sql(type, limit: nil, precision: nil, scale: nil, unsigned: nil, **) # :nodoc:
```
Comment (ColinDKelley) [inline lib/active_record/mysql/enum/mysql_adapter.rb]: Ditto re: `**_options`.

**Review by ColinDKelley — APPROVED** [review]
Looks good! Just a nit suggestion in two places.

**Comment by ColinDKelley** [comment]
@jebentier Maybe update the README in the is PR to explicitly mention Rails 4, 5, and 6 are all supported? Possibly the gemspec too?

File: lib/active_record/mysql/enum/mysql_adapter.rb (hunk 1 above)
Comment (jebentier) [reply lib/active_record/mysql/enum/mysql_adapter.rb]: done

File: lib/active_record/mysql/enum/mysql_adapter.rb (hunk 2 above)
Comment (jebentier) [reply lib/active_record/mysql/enum/mysql_adapter.rb]: done

**Comment by jebentier** [comment]
@ColinDKelley the README and gemspec already restrict to rails 5, which is all we've confirmed in our testing so far. There is a ticket that I've created to add modern unit tests to the project which will include figuring out rails 6 support.

**Pull request merged**